flask_admin/model/base.py

```diff
@@ -10,7 +10,7 @@
 from jinja2 import contextfunction
 from wtforms.fields import HiddenField
 from wtforms.fields.core import UnboundField
-from wtforms.validators import ValidationError, Required
+from wtforms.validators import ValidationError, InputRequired
```

Comment:
Yes, it does say that, and I'm not sure why it prescribes a single choice. From the docs for InputRequired:
"""
class wtforms.validators.InputRequired(message=None)
...
Note there is a distinction between this and DataRequired in that InputRequired looks that form-input data was provided, and DataRequired looks at the post-coercion data.
"""

In particular, if the input is for record id=0, the post-coercision data will evaluate to False. And so DataRequired would fail on this input, whereas InputRequired would succeed. Since we should allow operations on records with id=0, I believe InputRequired is the correct validator.

This stackoverflow clarified the difference better for me:
http://stackoverflow.com/questions/23982917/flask-wtforms-difference-between-datarequired-and-inputrequired